### PR TITLE
MS-628 Use the new LibSimprints data models if intent comes for latest LibSimprints version

### PR DESCRIPTION
**Pull request by luhmirin-s**
Updated SID to return data structures based on the caller's LibSimptints version. For simplicity moved the meta-data about the caller into `ActionRequest` class so that it is available throughout the execution flow.
 

**NOTE: This PR uses snapshot version of LibSimprints.**

### CODE REVIEW

File: feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/response/LibSimprintsResponseMapper.kt [hunk 1]
```diff
@@ -64,6 +112,16 @@ internal class LibSimprintsResponseMapper @Inject constructor() {
         )
     }
 
+    private inline fun Bundle.appendDataForVersion(
+        response: ActionResponse,
```
Comment (BurningAXE) [inline feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/response/LibSimprintsResponseMapper.kt:116]: TBH, I find it a little confusing that appendDataForVersion() does not take version as a parameter but actually returns it.

**Review by BurningAXE — APPROVED** [review]
Cannot say that I fully endorse the OldSomething naming but otherwise we'll have to deal with SomethingV1 and SomethingV2 which I don't like either :D 
However, the benefit of the latter is that if we end up changing the contract again, we won't end up with OlderSomething, OldOldSomething or NewSomething or some other abomination.

**Comment by luhmirin-s** [comment]
> Cannot say that I fully endorse the OldSomething naming but otherwise we'll have to deal with SomethingV1 and SomethingV2 which I don't like either :D However, the benefit of the latter is that if we end up changing the contract again, we won't end up with OlderSomething, OldOldSomething or NewSomething or some other abomination.

I will rename it to "Legacy..." to sound more serious. I would prefer to avoid the V1/2 at this time. It would cause even more confusion since we don't have anything that would match those versions. In the future versions we would have mechanisms to version the API calls in some way, so everything from before this new mechanism is a "legacy" implementation.

**Comment by alexandr-simprints** [comment]
What is the current state of this PR?

**Comment by luhmirin-s** [comment]
@alexandr-simprints It can be merged after sonar re-run, but I will do it after the 2024.2.1 is done, just in case.

File: infra/orchestrator-data/src/main/java/com/simprints/infra/orchestration/data/ActionRequestIdentifier.kt [hunk 2]
```diff
@@ -7,14 +7,21 @@ import kotlinx.parcelize.Parcelize
...
-        }
+        fun fromIntentAction(
+            action: String,
+            callerPackageName: String = "",
+            callerVersion: Int = 1,
+        ) = ActionRequestIdentifier(
+            actionName = action.substringAfterLast("."),
+            packageName = action.substringBeforeLast("."),
+            callerPackageName,
+            callerVersion,
```
[the remark this answers is not in the document]
Comment (luhmirin-s) [reply infra/orchestrator-data/src/main/java/com/simprints/infra/orchestration/data/ActionRequestIdentifier.kt:24]: Somehow I missed this comment. Makes sense 👍🏻

File: feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/response/LibSimprintsResponseMapper.kt (hunk 1 above)
Comment (luhmirin-s) [reply feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/response/LibSimprintsResponseMapper.kt:116]: Changed to `appendDataPerVersion` which is a bit more precise.

File: infra/orchestrator-data/src/main/java/com/simprints/infra/orchestration/data/ActionRequestIdentifier.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (BurningAXE) [reply infra/orchestrator-data/src/main/java/com/simprints/infra/orchestration/data/ActionRequestIdentifier.kt:24]: Not to be picky but maybe the parameter and variable can also be renamed for consistency.

**Pull request merged**